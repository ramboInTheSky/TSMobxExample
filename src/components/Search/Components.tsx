import styled from '@emotion/styled'
import Paper from '@material-ui/core/Paper'
import InputBase from '@material-ui/core/InputBase'
import IconButton from '@material-ui/core/IconButton'

export const Container: React.ComponentType<any> = styled(Paper)`
	display: flex;
	align-items: center;
`

export const Input: React.ComponentType<any> = styled(InputBase)`
    margin-left: 8;
    padding-left: 1rem; 
	flex: 1;
`

export const Action: React.ComponentType<any> = styled(IconButton)`padding: 10;`
