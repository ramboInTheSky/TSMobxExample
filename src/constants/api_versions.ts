export default Object.freeze({
    profile: 'ac-api/',
    visit: 'ac-api/',
    archetype: 'ac-api/',
    lockgroup: 'ac-api/',
})