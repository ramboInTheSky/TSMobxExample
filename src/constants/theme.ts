
export const variables = Object.freeze({
    columnSize: 80,
    componentBackground: '#ececec',
    componentBackgroundClear: '#fff',
    spacer: '2px solid #999',
    gutter: '1rem',
    indentation: '0.3rem',
    lineHeightComponentTitle: '2',
    headerFontSize: '1.2rem',
    archetypeColor: '#a4d8f1',
    tagColor: '#b4dec7',
    roomColor: '#eff180',
    primaryColor: '#b53f79',
    visitDateFromColor: '#999',
    visitDateToColor: '#999',
    dangerColor: '#f72b2b',
})