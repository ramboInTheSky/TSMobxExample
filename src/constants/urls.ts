import apiVersion from './api_versions'

export const urls = Object.freeze({
	profiles: `https://csi-dev.thecollective.com/${apiVersion.profile}profile`,
	visits: `https://csi-dev.thecollective.com/${apiVersion.visit}visit`,
	archetypes: `https://csi-dev.thecollective.com/${apiVersion.archetype}archetype`,
	lockGroups: `https://csi-dev.thecollective.com/${apiVersion.lockgroup}lockgroup`,
})
export default urls