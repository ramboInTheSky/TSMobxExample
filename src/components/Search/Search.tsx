import React from 'react'
import SearchIcon from '@material-ui/icons/Search'
import CloseIcon from '@material-ui/icons/Close'

import { Boundary } from '../ErrorBoundary'
import { Container, Input, Action } from './Components'

export interface SearchProps {
    onChange: (e: any, value?: string) => void
    placeholder?: string
}

export interface SearchState {
    value: string
}

export class Search extends React.Component<SearchProps, SearchState> {

    public state: SearchState = { value: '' }
    constructor(props: SearchProps) {
        super(props)
    }

    public clearSelection = () => {
        const { onChange } = this.props
        this.setState({ value: '' }, () => onChange({ target: { value: '' } }))
    }

    public render() {
        const { placeholder, onChange } = this.props
        const { value } = this.state
        return (
            <Boundary>
                <Container elevation={1}>
                    {value && (
                        <Action aria-label="Clear" onClick={this.clearSelection} title="Clear Search">
                            <CloseIcon />
                        </Action>
                    )}
                    <Input
                        placeholder={placeholder || 'Search: name, company, access, key number'}
                        onChange={(e: any) => this.setState({ value: e.target.value })}
                        value={value}
                    />

                    <Action aria-label="Search" onClick={(e: any) => onChange(e, value)} title="Search">
                        <SearchIcon />
                    </Action>
                </Container>
            </Boundary>
        )
    }
}
