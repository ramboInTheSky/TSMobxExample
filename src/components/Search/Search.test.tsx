import React from 'react'
import { shallow } from 'enzyme'
import { createShallow } from '@material-ui/core/test-utils';

import { Search, SearchProps } from './Search'
import { ELOOP } from 'constants';

describe('<Search />', () => {
	let props: SearchProps
	let shallowMaterial: any
	beforeEach(() => {
		props = {
            onChange: jest.fn(()=>{}),
            placeholder: 'this is a placeholder'
		}
		shallowMaterial = createShallow()
	})
	it('renders correctly', () => {
		expect(shallow(<Search {...props}/>)).toMatchSnapshot()
	})
	it('calls onChange on clicking clear', () => {
		const component = shallowMaterial(<Search {...props}/>)
		component.find('[title="Search"]').simulate('click')
		expect(props.onChange).toHaveBeenCalled()
	})
})
