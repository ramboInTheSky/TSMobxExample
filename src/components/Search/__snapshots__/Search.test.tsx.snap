// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Search /> renders correctly 1`] = `
<Boundary>
  <Styled(WithStyles(Paper))
    elevation={1}
  >
    <Styled(WithStyles(WithFormControlContext(InputBase)))
      onChange={[Function]}
      placeholder="this is a placeholder"
      value=""
    />
    <Styled(WithStyles(IconButton))
      aria-label="Search"
      onClick={[Function]}
      title="Search"
    >
      <pure(SearchIcon) />
    </Styled(WithStyles(IconButton))>
  </Styled(WithStyles(Paper))>
</Boundary>
`;
